### PR TITLE
BLD: unpin pyqt < 5.15

### PR DESCRIPTION
**Pull request by tangkong**
## Description
Try unpinning qtpy for future py3.11 compatibility

## Motivation and Context
Originally motivated by me trying to get atef tests to pass.

In the past we pinned qtpy >5, < 5.15 in a variety of repositories, but have lost track of the reasoning.  (segfaults?...)
Currently 3.11 builds fail because this constrains qtpy enough to have no py311 builds.  
 
## How Has This Been Tested?
Go for it GHA

## Where Has This Been Documented?
This PR

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by tangkong** [comment]
current py3.11 roadblock seems to be `matplotlib-base` not finding a compatible version.  `matplotlib-base>=3.7` seems to work fine, but the solver believes it's pinned to 3.5.2.  I'll revisit this later, but to root of the change (unpinning pyqt) seems to be fine

**Comment by ZLLentz** [comment]
I think this is a correct change regardless

**Comment by klauer** [comment]
https://github.com/slaclab/pydm/pull/849

**Comment by ZLLentz** [comment]
Ah, it was me! My worst nemesis!

**Comment by klauer** [comment]
And a follow-up removing the pin: https://github.com/slaclab/pydm/pull/999
(I wish it was one issue/PR higher, then it'd have been the first with 4 digits. Oh well)

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ requirements:
     - scipy
   run_constrained:
     - pmgr >=2.0.2
-    - pyqt <5.15
+    - pyqt
```
[the remark this answers is not in the document]
Comment (tangkong) [reply conda-recipe/meta.yaml:42]: You're right, we can just remove this.  I brain farted on `run_constrained`

[the remark this answers is not in the document]
Comment (klauer) [reply conda-recipe/meta.yaml]: `pyqt =5` might still be advisable, looking forward to if/when(?) pyqt6 gets on conda-forge. It's probably fine for now.

**Review by klauer — APPROVED** [review]

**Pull request merged**